src/api/app/models/user.rb

```diff
@@ -842,11 +842,11 @@ def to_param
     to_s
   end
 
-  def nr_of_requests_that_need_work
-    Rails.cache.fetch("requests_for_#{login}", expires_in: 2.minutes) do
-      BsRequest.collection(user: login, states: %w(declined), roles: %w(creator)).count +
-      BsRequest.collection(user: login, states: %w(new), roles: %w(maintainer)).count +
-      BsRequest.collection(user: login, roles: %w(reviewer), reviewstates: %w(new), states: %w(review)).count
+  def tasks
+    Rails.cache.fetch("requests_for_#{cache_key}") do
+      declined_requests.count +
+      incoming_requests.count +
+      involved_reviews.count
```

Comment:
> However, again, these are corner cases and just for counting. It is more important that the user sees all tasks. Is the review task listed in the users task list when he is creator and reviewer?

This was basically my question :smile: 
With the user task list you mean the ``Incoming Reivews`` table on the user page? In this table we do NOT display requests with the same creator and responsible reviewer, so that was basically the reason to also not count it into here and just reuse this functionality!

For the uniq: This is a valid point, however, was already implemented like this before. And as this function is quite "heavy" I would leave it as it is because the only way to get rid of duplicates would be in the ruby code which would be significantly slower than doing this on DB level (like here).